rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/fleet_manager.py

```diff
@@ -0,0 +1,259 @@
+            else:
+                data['data']['destination_arrival_duration'] = 0.0
+                data['data']['completed_request'] = True
+            return data
+
+        @app.post('/open-rmf/rmf_demos_fm/navigate/')
+        async def navigate(robot_name: str, dest: Request):
+            data = {'success': False, 'msg': ''}
+            if (robot_name not in self.robots or
+                    len(dest.destination) < 1):
+                return data
+
+            target_x = dest.destination['x']
+            target_y = dest.destination['y']
+            target_yaw = dest.destination['yaw']
+
+            t = self.get_clock().now().to_msg()
+
+            path_request = PathRequest()
+            state = self.robots[robot_name]
+            cur_x = state.state.location.x
+            cur_y = state.state.location.y
+            cur_yaw = state.state.location.yaw
+            cur_loc = state.state.location
+            path_request.path.append(cur_loc)
+
+            disp = self.disp([target_x, target_y], [cur_x, cur_y])
+            duration = int(disp/self.vehicle_traits.linear.nominal_velocity) +\
+                int(abs(abs(cur_yaw) - abs(target_yaw)) /
+                    self.vehicle_traits.rotational.nominal_velocity)
+            t.sec = t.sec + duration
+            target_loc = Location()
+            target_loc.t = t
+            target_loc.x = target_x
+            target_loc.y = target_y
+            target_loc.yaw = target_yaw
+            target_loc.level_name = state.state.location.level_name
+
+            path_request.fleet_name = self.fleet_name
+            path_request.robot_name = robot_name
+            path_request.path.append(target_loc)
+            self.task_id = self.task_id + 1
+            path_request.task_id = str(self.task_id)
+            self.path_pub.publish(path_request)
+
+            self.robots[robot_name].destination = target_loc
+
+            data['success'] = True
+            return data
+
+        @app.get('/open-rmf/rmf_demos_fm/stop_robot/')
+        async def stop(robot_name: str):
+            data = {'success': False, 'msg': ''}
+            if robot_name not in self.robots:
+                return data
+            path_request = PathRequest()
+            path_request.fleet_name = self.fleet_name
+            path_request.robot_name = robot_name
+            path_request.path = []
+            self.task_id = self.task_id + 1
+            path_request.task_id = str(self.task_id)
+            self.path_pub.publish(path_request)
+            data['success'] = True
+            return data
+
+        @app.post('/open-rmf/rmf_demos_fm/start_task/')
+        async def start_process(robot_name: str, task: Request):
+            data = {'success': False, 'msg': ''}
+            if (robot_name not in self.robots or
+                    len(task.task) < 1):
+                return data
+            # ------------------------ #
```

Comment:
Implementing this endpoint is crucial for cleaning and docking related tasks. Lets have this node subscribe to `/dock_summary` and store a dict of {dock_name: waypoints} as populated in the `DockSummary` msg. Apart from `robot_name`, we should also receive the `process_name`. This would be the `dock_name` in `RobotCommandHandle::dock(dock_name:str)` . So when we get a request at this endpoint, we lookup the corresponding waypoints for the `dock_name` and send a single `PathRequest` to `slotcar` with the `path` field populated with all the waypoints.

Reply:
Ah my bad, I was confused about the cleaning path, didn't realise it was stuck. Made the necessary changes, `fleet_manager` now sends out a `PathRequest` populated with the docking waypoints